### PR TITLE
#839: Introduce IteratorOfBytes and IteratorOfDoubles

### PR DESCRIPTION
**Pull request by proshin-roman**
PR for #839.

- added IteratorOfBytes and IteratorOfDoubles
- added tests for new classes
- refactored implementation of IterableOfBytes and IterableOfDoubles so that they use new classes
- refactored IterableOfFloats and IterableOfInts: got rid of redundant UncheckedScalar wrapping

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#863](https://github.com/yegor256/cactoos/issues/863) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#863](https://github.com/yegor256/cactoos/issues/863) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C63314D6Z/1361140393); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/java/org/cactoos/iterator/IteratorOfBytes.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ */
+package org.cactoos.iterator;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Iterator tat returns a set of bytes.
+ *
```
Comment (paulodamaso) [inline src/main/java/org/cactoos/iterator/IteratorOfBytes.java]: @proshin-roman Please correct this typo.

File: src/main/java/org/cactoos/iterator/IteratorOfDoubles.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ */
+package org.cactoos.iterator;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Iterator tat returns a set of double values.
+ *
```
Comment (paulodamaso) [inline src/main/java/org/cactoos/iterator/IteratorOfDoubles.java]: @proshin-roman Please correct this typo.

**Comment by proshin-roman** [comment]
@paulodamaso I've fixed all comments - please review the pull request again

**Comment by paulodamaso** [comment]
@rultor good to merge

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#863](https://github.com/yegor256/cactoos/issues/863) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@paulodamaso According to our Policy:
> The code reviewer found at least three problems in the code (discuss);
> Comments were mostly about design problems, not cosmetic issues;

Please, confirm that you will try to find at least 3 not cosmetic issues next time.

**Comment by elenavolokhova** [comment]
@0crat quality acceptable

**Comment by elenavolokhova** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/863#issuecomment-390326779))

@elenavolokhova This is what I know about this job, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#863](https://github.com/yegor256/cactoos/issues/863) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%23863%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Pull request merged**